driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -287,14 +314,13 @@ public static MongoCredential createGSSAPICredential(final String userName) {
     /**
      * Creates a MongoCredential instance for the MONGODB-AWS mechanism.
      *
-     * @param userName the user name
-     * @param password the user password
+     * @param userName the user name, which may be null.  This maps to the AWS accessKeyId
+     * @param password the user password, which may be null if the userName is also null.  This maps to the AWS secretAccessKey.
      * @return the credential
      * @since 4.1
      * @mongodb.server.release 4.4
-     * @see #createCredential(String, String, char[])
      */
-    public static MongoCredential createAwsCredential(final String userName, final char[] password) {
+    public static MongoCredential createAwsCredential(@Nullable final String userName, @Nullable final char[] password) {
```

Comment:
I added `@see` javadoc for `withMechanismProperty` as well as the two mechanism keys that you can use with an AWS credential.  The Javadoc for the `AWS_CREDENTIAL_PROVIDER_KEY` in turn links to `AwsCredential`, which seems a more appropriate place for that link since it's the only way to actually use `AwsCredential`.